src/group_theory/group_action/defs.lean

```diff
@@ -339,3 +358,39 @@ theorem smul_sub (r : M) (x y : A) : r • (x - y) = r • x - r • y :=
 by rw [sub_eq_add_neg, sub_eq_add_neg, smul_add, smul_neg]
 
 end
+
+/-- The tautological action by `function.End α` on `α`. -/
+instance mul_action.function_End : mul_action (function.End α) α :=
```

Comment:
Can you add the perm version of this instance too in `group_theory/perm`?

Reply:
Also, a `function.End.smul_def : f \smul x = f x` lemma (possibly simp) is probably a good idea.

Reply:
```suggestion
instance function.End.mul_action : mul_action (function.End α) α :=
```
might make the `function.End.smul_def` lemma easier to find